### ISSUE TITLE
Why does the Patience Parameter distort the training map scores in validation?

### ISSUE THREAD

**Issue by codejoetheduke** [ultralytics/ultralytics#20615]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained a yolo11l model on a custom dataset with 3 classes for 100 epochs and had a score of 0.821. The model stopped improving at the 46th epoch. Now when I added patience, I get a far different result and lower map50 score at validation. Isn't there a way of solving this? I want the same result but shortened to 46 epochs.

**Comment by Y-T-G** [comment]
What's the code?

**Comment by codejoetheduke** [comment]
# 🔧 Load Pretrained YOLOv11 Model
model1 = YOLO("yolo11l.pt")

# 🚀 Simulate 100-Epoch Training in 46 Epochs
model1.train(data='data.yaml',
             epochs=100,  # shortened to 46
             imgsz=640,
             device=0,
             batch=16,
             optimizer='AdamW',
             lr0=3e-4,           # initial learning rate (same as before)
             momentum=0.9,
             weight_decay=1e-2,
             close_mosaic=30,
             seed=42,
             patience=10          # disable early stopping (optional but recommended)
)

**Comment by codejoetheduke** [comment]
This gave a different result from:

model1 = YOLO("yolo11l.pt")

model1.train(data='data.yaml',
             epochs=100,  
             imgsz=640,
             device=0,
             batch=16,
             optimizer='AdamW',
             lr0=3e-4,           # initial learning rate (same as before)
             momentum=0.9,
             weight_decay=1e-2,
             close_mosaic=0,
             seed=42,
)

**Comment by codejoetheduke** [comment]
Yes please. The whole log is different

**Comment by codejoetheduke** [comment]
If better, is there another method that will take the same steps as the 100 epochs would do but stop at 46 epochs of training?

**Comment by Y-T-G** [comment]
I can't reproduce it. It's the same metrics with different patience values. Are you using the latest Ultralytics?

**Comment by codejoetheduke** [comment]
I am doing it a custom dataset. I guess so. I am using a kaggle notebook.

**Comment by codejoetheduke** [comment]
I am also quite confused. So the default patience value is 100 per the documentation. Does that mean if I change the value to 10 it distorts the results? How can I solve this then?

**Comment by codejoetheduke** [comment]
I am re running my experiment. I think I might have made a mistake. Just a minute. Thank you for patience sir.

**Comment by codejoetheduke** [comment]
@Y-T-G It is working perfectly now. Thank you and sorry for the disturbance. My bad 😅
